Objects/unionobject.c

```diff
@@ -368,6 +387,151 @@ _Py_union_args(PyObject *self)
     return ((unionobject *) self)->args;
 }
 
+static PyObject *
+union_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+    if (!_PyArg_NoKeywords("UnionType", kwds)) {
+        return NULL;
+    }
+    return _Py_union_from_tuple(args);
+}
+
+static PyObject *
+call_typing_func_object(const char *name, PyObject **args, size_t nargs)
+{
+    PyObject *typing = PyImport_ImportModule("typing");
+    if (typing == NULL) {
+        return NULL;
+    }
+    PyObject *func = PyObject_GetAttrString(typing, name);
+    if (func == NULL) {
+        Py_DECREF(typing);
+        return NULL;
+    }
+    PyObject *result = PyObject_Vectorcall(func, args, nargs, NULL);
+    Py_DECREF(func);
+    Py_DECREF(typing);
+    return result;
+}
+
+static PyObject *
+type_check(PyObject *arg, const char *msg)
+{
+    // Calling typing.py here leads to bootstrapping problems
+    if (Py_IsNone(arg)) {
+        return Py_NewRef(Py_TYPE(arg));
```

Comment:
`PyNone_Type` is immortal (all static types are), so I don't think the `Py_NewRef` is needed here